server/datastore/mysql/software.go

```diff
@@ -2375,7 +2375,22 @@ INNER JOIN software_cve scve ON scve.software_id = s.id
 					hvsi.removed = 0
 			) AND
 			-- either the software installer or the vpp app exists for the host's team
-			( si.id IS NOT NULL OR vat.platform = :host_platform )
+			( si.id IS NOT NULL OR vat.platform = :host_platform ) AND
+			-- label membership check
+			(
+				NOT EXISTS (
+					SELECT
+						1 FROM software_installer_labels sil JOIN label_membership lm ON lm.label_id = sil.label_id AND lm.host_id = :host_id
+					WHERE
+						sil.software_installer_id = si.id
+						AND sil.exclude = 1)
+				OR EXISTS (
+					SELECT
+						1 FROM software_installer_labels sil JOIN label_membership lm ON lm.label_id = sil.label_id AND lm.host_id = :host_id
+					WHERE
+						sil.software_installer_id = si.id
+						AND sil.exclude = 0)
```

Comment:
Would something simpler like this work? Trying to reduce 2 subquery/loops to just 1.

```sql
    EXISTS (
    SELECT 1 FROM software_installer_labels sil
        LEFT JOIN label_membership lm ON lm.label_id = sil.label_id AND lm.host_id = :host_id
    WHERE lm.label_id IS NULL OR (lm.label_id IS NOT NULL AND sil.exclude = 0)
    )
```

Reply:
@getvictor yeah I've been chewing on this issue this afternoon, I think I just came up with something that will be simpler! Will push soon.